### ISSUE TITLE
Extension and zc.buildout/setuptools version pin raises conflict

### ISSUE THREAD

**Issue by PCatinean** [buildout/buildout#381]
Any buildout that has an extension cannot pin the versions of the zc.buildout or setuptools as it raises a conflict:

```[buildout]

parts = test

extensions = gp.vcsdevelop
vcs-extend-develop = something
vcs-update = True

[test]
recipe = some recipe

[versions]
zc.buildout = 2.9.0
setuptools = 19.7```

Is there no way to force the order in which the versions are pulled?

It seems the having an extension uses the latest buildout and then you cannot downgrade. While if you just remove the extension it works and pins the version properly

**Comment by PCatinean** [comment]
@reinout I noticed you discussed some similar topics. Any idea what a good workaround for this is?

**Comment by jimfulton** [comment]
See: https://groups.google.com/forum/#!topic/buildout-development/bfGPpgirAhk

IOW, if you get a conflict, try:

    bin/buildout extensions=

to get buildout upgraded then run buildout as usual.

I'm thinking that we should delay loading extensions until after upgrading.

**Comment by PCatinean** [comment]
@jimfulton not sure how the extension might help.

I mean in the end if I run it a few more times it eventually works, but that's very unreliable and bad.

Indeed delaying extension loading after upgrade is a good approach.

Did you try to start a minimal buildout such as that one and pin an earlier version of buildout or setuptools?

**Comment by jimfulton** [comment]
No, I just happened to run into this yesterday on work for a customer. :)

Bonus: https://github.com/pypa/setuptools/issues/997

I haven't used any extensions in quite a while, so this hasn't bitten me, but I want/need to start using the new wheel extension.

**Comment by livingmine** [comment]
@PCatinean Is this the error you got?
```
Getting distribution for 'gp.vcsdevelop'.
warning: no previously-included files matching '*' found under directory 'docs/_build'
Got gp.vcsdevelop 2.2.3.
Getting distribution for 'zc.buildout==2.5.2'.
While:
  Installing.
  Loading extensions.
  Getting distribution for 'zc.buildout==2.5.2'.

An internal error occurred due to a bug in either zc.buildout or in a
recipe being used:
```
I encountered the above error, strangely if i run the ```bin/buildout -c buildout.cfg``` over and over again without changing anything, eventually it works just fine.

**Comment by PCatinean** [comment]
@livingmine same behavior, it's just annoying to do it many times over. I patched up my local bootstrap.py to make checks before getting zc.buildout and setuptools. No permanent solution other specifying the versions yourself using the command line.
